tests/integration/test_kube_ovn.py

```diff
@@ -445,6 +447,38 @@ async def test_bgp(ops_test, ips_to_curl):
             assert await run_bird_curl_test(ops_test, unit, ip)
 
 
+async def test_network_policies(ops_test, client, kubectl_exec, network_policies):
+    blocked_pod, allowed_pod = network_policies
+
+    @retry(
+        retry=retry_if_exception_type(AssertionError),
+        stop=stop_after_delay(600),
+        wait=wait_fixed(1),
+        before=before_log(log, logging.INFO),
+    )
+    async def check_wget(url, client, msg):
+        stdout = await wget(kubectl_exec, client, url)
+        assert msg in stdout
+
+    log.info("Checking pods connectivity...")
+    for pod in network_policies:
+        await check_wget("nginx.netpolicy", pod, "'index.html' saved")
+
+    log.info("Applying NetworkPolicy...")
+    path = Path("tests/data/net-policy.yaml")
+    policies = load_all_yaml(path.read_text())
+    for obj in policies:
+        client.create(obj)
+
+    log.info("Checking NetworkPolicy...")
+    await check_wget("nginx.netpolicy", allowed_pod, "'index.html' saved")
+    await check_wget("nginx.netpolicy", blocked_pod, "wget: download timed out")
+
+    log.info("Removing NetworkPolicy...")
+    for obj in policies:
+        client.delete(type(obj), obj.metadata.name, obj.metadata.namespace)
```

Comment:
may i suggest this alternative as a way to still teardown even if the checks in the `try` block fail

```suggestion
    try:
        log.info("Checking NetworkPolicy...")
        await check_wget("nginx.netpolicy", allowed_pod, "'index.html' saved")
        await check_wget("nginx.netpolicy", blocked_pod, "wget: download timed out")
    finally:
        log.info("Removing NetworkPolicy...")
        for obj in policies:
            client.delete(type(obj), obj.metadata.name, obj.metadata.namespace)
```

Reply:
Thanks @addyess, I will commit your suggestion straight away 😊